srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/NewRuleController.java

```diff
@@ -0,0 +1,136 @@
+    protected static final String ATTRIBUTE_RULE = "rule";
+    
+    private final AdminService fAdminService;
+    
+    /**
+     * Constructs an instance.
+     * @param adminService the AdminService to use for operations
+     * @throws NullPointerException if adminService is null
+     */
+    @Inject
+    public NewRuleController(final AdminService adminService)
+    {
+        fAdminService = Objects.requireNonNull(adminService);
+    }
+    
+    @ModelAttribute(ATTRIBUTE_RULE)
+    private EditRule createEditRule()
+    {
+        fLogger.trace("Creating EditDiscreteNumericalVar.");
+        return new EditRule(fAdminService);
+    }
+    
+    @RequestMapping(value = NEW_RULE_URL, method = RequestMethod.GET)
+    public ModelAndView displayForm(
+            @ModelAttribute(ATTRIBUTE_RULE) final EditRule editRule) throws InvalidIdentifierException
+    {
+        final ModelAndView mav = new ModelAndView(Views.EDIT_RULE);
+        // Provide the URL to save the rule to the view.
+        mav.addObject("SAVE_URL", NEW_RULE_URL);
+        // Note: "rule" is already in the model via createEditRule().
+        mav.addObject("isNewRule", true);
+        // A map of the variable's key to its respective summary
+        final Map<String, VariableSummary> variableSummaries = new HashMap<String, VariableSummary>();
+        for(final ValueMatcherBuilder matcher: editRule.getMatchers())
+        {
+            final Variable var = fAdminService.getVariable(matcher.getVariableKey());
+            variableSummaries.put(matcher.getVariableKey(), VariableSummary.fromVariable(var));
+        }
+        mav.addObject("variableSummaries", variableSummaries);
+        mav.addObject("allVariables", fAdminService.getAllVariables());
+        return mav;
+    }
+    
+    @RequestMapping(value = NEW_RULE_URL, method = RequestMethod.POST)
+    public ModelAndView saveRule(
+            @ModelAttribute(ATTRIBUTE_RULE) final EditRule editRule,
+            final BindingResult bindingResult,
+            @RequestParam(value = "submitButton") final String submitString) throws InvalidIdentifierException
+    {
+        if(submitString.equals("newMatcher"))
+        {
+            // Reload the page with the new matcher builder inside of the edit rule
+            editRule.getMatchers().add(
+                    new ValueMatcherBuilder().setVariableKey(editRule.getNewVariableKey()));
+            
+            return displayForm(editRule);
+        }
+        // Test if a remove button was clicked, and remove the proper ValueMatcherBuilder if needed
+        final Integer removeIndex = Ints.tryParse(submitString);
+        if(removeIndex != null)
+        {
+            editRule.getMatchers().remove(removeIndex.intValue());
+            return displayForm(editRule);
+        }
+        // Call the validator for an EditRule here so that we can access the editRule
+        // variable to display 
+        editRule.getValidator().validate(editRule, bindingResult);
```

Comment:
Did you intentionally only call the validator if we're actually trying to save the rule? I like that.